Generic type parameter causes java library type resolution to fail

The following Java program:

```
/*@
pure boolean f<T>(seq<T> xs) = true;
@*/
```

Causes the following error:

```
[INFO] Starting verification
[ERROR] Unrecoverable error: Cannot invoke "java.io.File.toPath()" because "baseFile" is null
java.lang.NullPointerException: Cannot invoke "java.io.File.toPath()" because "baseFile" is null
        at vct.col.resolve.lang.Java$.$anonfun$findLibraryJavaType$9(Java.scala:192)
        at scala.Option.map(Option.scala:242)
        at vct.col.resolve.lang.Java$.$anonfun$findLibraryJavaType$6(Java.scala:188)
        at scala.Option.flatMap(Option.scala:283)
        at vct.col.resolve.lang.Java$.$anonfun$findLibraryJavaType$5(Java.scala:187)
        at scala.Option$WithFilter.flatMap(Option.scala:283)
        at vct.col.resolve.lang.Java$.$anonfun$findLibraryJavaType$3(Java.scala:186)
        at scala.Option.flatMap(Option.scala:283)
        at vct.col.resolve.lang.Java$.$anonfun$findLibraryJavaType$2(Java.scala:185)
        at hre.util.FuncTools$.$anonfun$firstOption$4(FuncTools.scala:11)
        at scala.Option.orElse(Option.scala:477)
        at hre.util.FuncTools$.$anonfun$firstOption$3(FuncTools.scala:11)
        at scala.collection.LinearSeqOps.foldLeft(LinearSeq.scala:169)
        at scala.collection.LinearSeqOps.foldLeft$(LinearSeq.scala:165)
        at scala.collection.immutable.List.foldLeft(List.scala:79)
        at hre.util.FuncTools$.firstOption(FuncTools.scala:11)
        at vct.col.resolve.lang.Java$.$anonfun$findLibraryJavaType$1(Java.scala:178)
        at scala.Option.flatMap(Option.scala:283)
        at vct.col.resolve.lang.Java$.findLibraryJavaType(Java.scala:177)
        at vct.col.resolve.lang.Java$.$anonfun$findJavaTypeName$3(Java.scala:232)
        at hre.util.FuncTools$.$anonfun$firstOption$4(FuncTools.scala:11)
        at scala.Option.orElse(Option.scala:477)
        at hre.util.FuncTools$.$anonfun$firstOption$3(FuncTools.scala:11)
        at scala.collection.LinearSeqOps.foldLeft(LinearSeq.scala:169)
        at scala.collection.LinearSeqOps.foldLeft$(LinearSeq.scala:165)
        at scala.collection.immutable.List.foldLeft(List.scala:79)
        at hre.util.FuncTools$.firstOption(FuncTools.scala:11)
        at vct.col.resolve.lang.Java$.$anonfun$findJavaTypeName$2(Java.scala:232)
        at scala.Option.orElse(Option.scala:477)
        at vct.col.resolve.lang.Java$.findJavaTypeName(Java.scala:232)
        at vct.col.resolve.ResolveTypes$.resolveOne(Resolve.scala:43)
        at vct.col.resolve.ResolveTypes$.resolve(Resolve.scala:27)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1$adapted(Resolve.scala:26)
        at scala.collection.immutable.List.foreach(List.scala:333)
        at vct.col.resolve.ResolveTypes$.resolve(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1$adapted(Resolve.scala:26)
        at scala.collection.immutable.List.foreach(List.scala:333)
        at vct.col.resolve.ResolveTypes$.resolve(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1$adapted(Resolve.scala:26)
        at scala.collection.immutable.List.foreach(List.scala:333)
        at vct.col.resolve.ResolveTypes$.resolve(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1$adapted(Resolve.scala:26)
        at scala.collection.immutable.List.foreach(List.scala:333)
        at vct.col.resolve.ResolveTypes$.resolve(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1$adapted(Resolve.scala:26)
        at scala.collection.immutable.List.foreach(List.scala:333)
        at vct.col.resolve.ResolveTypes$.resolve(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.resolve(Resolve.scala:20)
        at vct.main.stages.Resolution.run(Resolution.scala:66)
        at vct.main.stages.Resolution.run(Resolution.scala:42)
        at hre.stages.UnitStages.runUnsafely(Stages.scala:42)
        at hre.stages.StagesPair.runUnsafely(Stages.scala:50)
        at hre.stages.StagesPair.runUnsafely(Stages.scala:48)
        at hre.stages.StagesPair.runUnsafely(Stages.scala:48)
        at hre.stages.StagesPair.runUnsafely(Stages.scala:48)
        at hre.stages.Stages.$anonfun$run$1(Stages.scala:27)
        at hre.progress.Progress$.withFrame(Progress.scala:146)
        at hre.progress.Progress$.stages(Progress.scala:154)
        at hre.stages.Stages.run(Stages.scala:27)
        at hre.stages.Stages.run$(Stages.scala:24)
        at hre.stages.StagesPair.run(Stages.scala:46)
        at vct.main.modes.Verify$.verifyWithOptions(Verify.scala:40)
        at vct.main.modes.Verify$.runOptions(Verify.scala:68)
        at vct.main.Main$.runOptions(Main.scala:80)
        at vct.main.Main$.main(Main.scala:38)
        at vct.main.Main.main(Main.scala)
Exception in thread "main" java.lang.NullPointerException: Cannot invoke "java.io.File.toPath()" because "baseFile" is null
        at vct.col.resolve.lang.Java$.$anonfun$findLibraryJavaType$9(Java.scala:192)
        at scala.Option.map(Option.scala:242)
        at vct.col.resolve.lang.Java$.$anonfun$findLibraryJavaType$6(Java.scala:188)
        at scala.Option.flatMap(Option.scala:283)
        at vct.col.resolve.lang.Java$.$anonfun$findLibraryJavaType$5(Java.scala:187)
        at scala.Option$WithFilter.flatMap(Option.scala:283)
        at vct.col.resolve.lang.Java$.$anonfun$findLibraryJavaType$3(Java.scala:186)
        at scala.Option.flatMap(Option.scala:283)
        at vct.col.resolve.lang.Java$.$anonfun$findLibraryJavaType$2(Java.scala:185)
        at hre.util.FuncTools$.$anonfun$firstOption$4(FuncTools.scala:11)
        at scala.Option.orElse(Option.scala:477)
        at hre.util.FuncTools$.$anonfun$firstOption$3(FuncTools.scala:11)
        at scala.collection.LinearSeqOps.foldLeft(LinearSeq.scala:169)
        at scala.collection.LinearSeqOps.foldLeft$(LinearSeq.scala:165)
        at scala.collection.immutable.List.foldLeft(List.scala:79)
        at hre.util.FuncTools$.firstOption(FuncTools.scala:11)
        at vct.col.resolve.lang.Java$.$anonfun$findLibraryJavaType$1(Java.scala:178)
        at scala.Option.flatMap(Option.scala:283)
        at vct.col.resolve.lang.Java$.findLibraryJavaType(Java.scala:177)
        at vct.col.resolve.lang.Java$.$anonfun$findJavaTypeName$3(Java.scala:232)
        at hre.util.FuncTools$.$anonfun$firstOption$4(FuncTools.scala:11)
        at scala.Option.orElse(Option.scala:477)
        at hre.util.FuncTools$.$anonfun$firstOption$3(FuncTools.scala:11)
        at scala.collection.LinearSeqOps.foldLeft(LinearSeq.scala:169)
        at scala.collection.LinearSeqOps.foldLeft$(LinearSeq.scala:165)
        at scala.collection.immutable.List.foldLeft(List.scala:79)
        at hre.util.FuncTools$.firstOption(FuncTools.scala:11)
        at vct.col.resolve.lang.Java$.$anonfun$findJavaTypeName$2(Java.scala:232)
        at scala.Option.orElse(Option.scala:477)
        at vct.col.resolve.lang.Java$.findJavaTypeName(Java.scala:232)
        at vct.col.resolve.ResolveTypes$.resolveOne(Resolve.scala:43)
        at vct.col.resolve.ResolveTypes$.resolve(Resolve.scala:27)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1$adapted(Resolve.scala:26)
        at scala.collection.immutable.List.foreach(List.scala:333)
        at vct.col.resolve.ResolveTypes$.resolve(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1$adapted(Resolve.scala:26)
        at scala.collection.immutable.List.foreach(List.scala:333)
        at vct.col.resolve.ResolveTypes$.resolve(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1$adapted(Resolve.scala:26)
        at scala.collection.immutable.List.foreach(List.scala:333)
        at vct.col.resolve.ResolveTypes$.resolve(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1$adapted(Resolve.scala:26)
        at scala.collection.immutable.List.foreach(List.scala:333)
        at vct.col.resolve.ResolveTypes$.resolve(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.$anonfun$resolve$1$adapted(Resolve.scala:26)
        at scala.collection.immutable.List.foreach(List.scala:333)
        at vct.col.resolve.ResolveTypes$.resolve(Resolve.scala:26)
        at vct.col.resolve.ResolveTypes$.resolve(Resolve.scala:20)
        at vct.main.stages.Resolution.run(Resolution.scala:66)
        at vct.main.stages.Resolution.run(Resolution.scala:42)
        at hre.stages.UnitStages.runUnsafely(Stages.scala:42)
        at hre.stages.StagesPair.runUnsafely(Stages.scala:50)
        at hre.stages.StagesPair.runUnsafely(Stages.scala:48)
        at hre.stages.StagesPair.runUnsafely(Stages.scala:48)
        at hre.stages.StagesPair.runUnsafely(Stages.scala:48)
        at hre.stages.Stages.$anonfun$run$1(Stages.scala:27)
        at hre.progress.Progress$.withFrame(Progress.scala:146)
        at hre.progress.Progress$.stages(Progress.scala:154)
        at hre.stages.Stages.run(Stages.scala:27)
        at hre.stages.Stages.run$(Stages.scala:24)
        at hre.stages.StagesPair.run(Stages.scala:46)
        at vct.main.modes.Verify$.verifyWithOptions(Verify.scala:40)
        at vct.main.modes.Verify$.runOptions(Verify.scala:68)
        at vct.main.Main$.runOptions(Main.scala:80)
        at vct.main.Main$.main(Main.scala:38)
        at vct.main.Main.main(Main.scala)
```

When `seq<T>` is changed into `seq<int>`, the error disappears.

--- a/build.sbt
+++ b/build.sbt
@@ -4,6 +4,10 @@ import sbt.internal._
 
 ThisBuild / turbo := true // en wat is daar het praktisch nut van?
 ThisBuild / scalaVersion := "2.13.5"
+ThisBuild / javacOptions :=Seq("-source", "17",
+                               "-target", "17",
+                               "-Xlint:deprecation", "-Xlint:unchecked",
+                               "-deprecation")
 ThisBuild / fork := true
 
 ThisBuild / pushRemoteCacheTo := Some(MavenCache("local-cache", file("tmp/vercors-build-cache")))
@@ -121,6 +125,7 @@ lazy val vercors: Project = (project in file("."))
     libraryDependencies += "org.jacoco" % "org.jacoco.agent" % "0.8.7" classifier "runtime",
 
     ThisBuild / scalacOptions ++= Seq(
+      "-target:jvm-1.17",
       "-deprecation",
       "-feature",
       "-unchecked",
@@ -134,24 +139,6 @@ lazy val vercors: Project = (project in file("."))
 //      "-P:scalac-profiling:show-profiles",
 //      "-P:scalac-profiling:sourceroot:/home/pieter/vercors/",
 //      "-Ypatmat-debug",
-    ),
-
-    Compile / javacOptions ++= Seq(
-      "-Xlint:deprecation",
-      "-Xlint:unchecked",
-      "-deprecation"
-    ),
-
-    Runtime / javacOptions ++= Seq(
-      "-Xlint:deprecation",
-      "-Xlint:unchecked",
-      "-deprecation"
-    ),
-
-    Test / javacOptions ++= Seq(
-      "-Xlint:deprecation",
-      "-Xlint:unchecked",
-      "-deprecation"
     ),
 
     buildInfoKeys := Seq[BuildInfoKey](name, version, scalaVersion, sbtVersion,

--- a/col/src/main/java/vct/col/resolve/lang/Java.scala
+++ b/col/src/main/java/vct/col/resolve/lang/Java.scala
@@ -264,11 +264,20 @@ case object Java extends LazyLogging {
       case moreNames => Seq(moreNames)
     }
 
-    FuncTools.firstOption(potentialFQNames, findLoadedJavaTypeName[G](_, ctx))
+
+
+    FuncTools.firstOption(potentialFQNames, findJavaTypeInStack[G](_, ctx))
+      .orElse(FuncTools.firstOption(potentialFQNames, findLoadedJavaTypeName[G](_, ctx)))
       .orElse(FuncTools.firstOption(potentialFQNames, findLibraryJavaType[G](_, ctx)))
       .orElse(FuncTools.firstOption(potentialFQNames, findRuntimeJavaType[G](_, ctx)).map(RefJavaClass[G]))
   }
 
+
+  def findJavaTypeInStack[G](name: Seq[String], ctx: TypeResolutionContext[G]): Option[JavaTypeNameTarget[G]] = {
+      ctx.stack.flatten.collectFirst{
+        case ref: JavaTypeNameTarget[G] if Seq(ref.name) == name => ref
+      }
+  }
 
   def findJavaName[G](name: String, ctx: TypeResolutionContext[G]): Option[JavaNameTarget[G]] = {
     ctx.stack.flatten.collectFirst {

